great_expectations/rule_based_profiler/util.py

```diff
@@ -54,8 +53,8 @@ def get_validator(
 
 
 def get_batch_ids(
-    data_context: Optional[DataContext] = None,
-    batch_request: Optional[Union[BatchRequest, dict, str]] = None,
+    data_context: Optional["DataContext"] = None,  # noqa: F821
+    batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
```

Comment:
@cdkini Where there used to be str as a type for batch_request it must remain. This is used for obtaining it from variables. Could you please restore it -- in addition to the other types, (BatchRequest, RuntimeBatchRequest, dict), wherever it used to be? Thanks!